typer/rich_utils.py

```diff
@@ -59,6 +59,7 @@
 STYLE_COMMANDS_TABLE_LEADING = 0
 STYLE_COMMANDS_TABLE_PAD_EDGE = False
 STYLE_COMMANDS_TABLE_PADDING = (0, 1)
+STYLE_COMMANDS_TABLE_EXPAND = True
```

Comment:
Having this as a separate option that by default behaves as before, is in itself a clean solution!